### PR TITLE
Implement `From<TextureFormat>` for `TextureComponentType`

### PR DESCRIPTION
**Pull request by mitchmindtree**

### CODE REVIEW

File: wgpu-types/src/lib.rs [hunk 1]
```diff
@@ -934,5 +934,52 @@ pub enum TextureComponentType {
     Uint,
 }
 
+impl From<TextureFormat> for TextureComponentType {
+    fn from(format: TextureFormat) -> Self {
+        match format {
+            TextureFormat::R8Unorm
+            | TextureFormat::R8Uint
```
Comment (kvark) [inline wgpu-types/src/lib.rs:941]: Is that how rustfmt formats this? I wish `TextureFormat` was aligned

File: wgpu-types/src/lib.rs [hunk 2]
```diff
@@ -934,5 +934,52 @@ pub enum TextureComponentType {
...
+            | TextureFormat::R16Uint
+            | TextureFormat::Rg8Unorm
```
Comment (kvark) [inline wgpu-types/src/lib.rs:943]: normalized formats (both unorm and snorm) should map to `Float`

File: wgpu-types/src/lib.rs (hunk 1 above)
Comment (mitchmindtree) [reply wgpu-types/src/lib.rs]: Indeed it does, if I add a `|` before the first pattern to align them, rustfmt removes it.

File: wgpu-types/src/lib.rs (hunk 2 above)
Comment (mitchmindtree) [reply wgpu-types/src/lib.rs]: Ahhh I see! I've added a commit that makes this change.

File: wgpu-types/src/lib.rs (hunk 1 above)
Comment (kvark) [reply wgpu-types/src/lib.rs]: what if you write it like `TextureFormat::xxx |` instead? Does rustfmt changes that too?

Comment (mitchmindtree) [reply wgpu-types/src/lib.rs]: Yes it does

**Comment by mitchmindtree** [comment]
Done!

For future reference, I *think* maintainers can auto-squash the commits of a PR via the drop-down-list next to the "Merge pull request" button. At least I think you still can, it's been a while since I've done it!

**Comment by kvark** [comment]
Yes, we'd totally do it ourselves if we could. But we use `bors` for this project, and it doesn't know how to squash. Thank you!
bors r+

**Pull request merged**